portal-api/src/cron-scheduler-jobs/helpers/utilisation-report-helpers.test.ts

```diff
@@ -0,0 +1,637 @@
+
+      const bankWithoutPaymentOfficerTeamEmail: BankResponse = {
+        ...aBank(),
+        isVisibleInTfmUtilisationReports: true,
+        paymentOfficerTeam: { teamName: 'Team' },
+      } as unknown as BankResponse;
+      jest.mocked(api.getAllBanks).mockResolvedValue([bankWithoutPaymentOfficerTeamEmail]);
+
+      jest.mocked(api.getUtilisationReports).mockResolvedValue([aNotReceivedUtilisationReportResponse()]);
+
+      const sendEmailCallback = jest.fn();
+
+      // Act
+      await sendEmailToAllBanksWhereReportNotReceived({
+        emailDescription: '',
+        sendEmailCallback,
+      });
+
+      // Assert
+      expect(sendEmailCallback).not.toHaveBeenCalled();
+      expect(console.warn).toHaveBeenCalledWith(expect.stringContaining('paymentOfficerTeam.emails property against bank is not an array or is empty'));
+    });
+
+    it('does not send an email when the bank has an invalid payment officer team email', async () => {
+      // Arrange
+      jest.mocked(externalApi.bankHolidays).getBankHolidayDatesForRegion.mockResolvedValue([]);
+
+      const invalidEmail = 'invalid-email';
+      const bank = aBank();
+      bank.isVisibleInTfmUtilisationReports = true;
+      bank.paymentOfficerTeam.emails = [invalidEmail];
+      jest.mocked(api.getAllBanks).mockResolvedValue([bank]);
+
+      jest.mocked(api.getUtilisationReports).mockResolvedValue([aNotReceivedUtilisationReportResponse()]);
+
+      const sendEmailCallback = jest.fn();
+
+      // Act
+      await sendEmailToAllBanksWhereReportNotReceived({
+        emailDescription: '',
+        sendEmailCallback,
+      });
+
+      // Assert
+      expect(sendEmailCallback).not.toHaveBeenCalled();
+      expect(console.error).toHaveBeenCalledWith(expect.stringContaining('invalid payment officer email'));
+    });
+
+    it('does not send an email when the bank is marked as not visible in TFM Utilisation Reports', async () => {
+      // Arrange
+      jest.mocked(externalApi.bankHolidays).getBankHolidayDatesForRegion.mockResolvedValue([]);
+
+      const bank: BankResponse = { ...aBank(), isVisibleInTfmUtilisationReports: false };
+      jest.mocked(api.getAllBanks).mockResolvedValue([bank]);
+
+      const sendEmailCallback = jest.fn();
+
+      // Act
+      await sendEmailToAllBanksWhereReportNotReceived({
+        emailDescription: '',
+        sendEmailCallback,
+      });
+
+      // Assert
+      expect(sendEmailCallback).not.toHaveBeenCalled();
+    });
+
+    it('sends emails using the default team name when bank does not have one set', async () => {
+      // Arrange
+      jest.mocked(externalApi.bankHolidays).getBankHolidayDatesForRegion.mockResolvedValue([]);
+
+      const validEmail = 'valid-email@example.com';
+      const bank = aBank();
+      bank.isVisibleInTfmUtilisationReports = true;
+      bank.paymentOfficerTeam = { emails: [validEmail] } as PaymentOfficerTeam;
```

Comment:
Is there any reason why we can't do `bank.paymentOfficerTeam.emails = [validEmail]` to avoid the type assertion?

Reply:
This line is also ensuring there is no team name provided, which is why we need the type assertion. It's a bit tricky to try and type that explicitly in a way I can trick typescript into thinking the bank is a Bank, so I will add a comment noting that is intentional.

Reply:
Ah I see. In that case, I generally prefer to use a `// @ts-expect-error <explanantion>` comment to highlight why typescript doesn't like it as you both get to avoid the `as` type assertion and get a comment explaining why it seems wrong